src/core/actions/cityactions/ResourceGathering.java

```diff
@@ -23,20 +24,69 @@ public ResourceGathering(City c)
 
     @Override
     public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO: compute resource gathering
-        return null;
+        LinkedList<Action> resources = new LinkedList<>();
+        if(isFeasible(gs)){
+            ResourceGathering resourceGathering = new ResourceGathering(this.city);
+            resourceGathering.setResource(this.resource);
+            resources.add(resourceGathering);
+        }
+        return resources;
     }
 
     @Override
     public boolean isFeasible(final GameState gs)
```

Comment:
This method should include a check on the cost of gathering this.resource (if the tribe of this city has enough stars)